src/RA_Dlg_MemBookmark.cpp

```diff
@@ -705,10 +709,10 @@ void Dlg_MemBookmark::ExportJSON()
 
     CComPtr<IFileSaveDialog> pDlg;
 
-    HRESULT hr;
+    auto hr{ S_OK };
```

Comment:
Why change this? It was very clearly an HRESULT before, now anyone reading the code has to rely on an external tool (i.e. Visual Studio) to identify the type. Additionally, it's immediately assigned a value on the next line, so why waste time setting it here?

Reply:
It has to be initilized something since it's a primitive.
Does HRESULT hr{}; work for you?

Reply:
Yes, but it's given an actual value on the next line. There's no need to explicitly initialize it here.
```
HRESULT hr;
if (SUCCEEDED(hr = doSomething()))
{
```
I hate to suggest putting hr in the if statement, but you wouldn't initialize it there.
```
if (HRESULT hr; SUCCEEDED(hr = doSomething()))
{
```

Reply:
I got you. It makes sense unlike the switch statements.